core/src/main/scala/com/databricks/labs/remorph/RootTableIdentifier.scala

```diff
@@ -0,0 +1,51 @@
+package com.databricks.labs.remorph
+
+import com.databricks.labs.remorph.parsers.{intermediate => ir}
+import com.databricks.labs.remorph.transpilers.TranspilerFactory
+
+class RootTableIdentifier(engine: String) {
+
+  private val transpiler = TranspilerFactory.getTranspiler(engine)
+
+  def processQuery(sql: String, dag: DAG): DAG = {
+    val parsedSql = transpiler.parse(sql)
+    findTables(dag, parsedSql)
+
+  }
+
+  private def getTablName(p: ir.LogicalPlan): Set[String] = {
+    p.find(_.isInstanceOf[ir.NamedTable]).map(_.asInstanceOf[ir.NamedTable].unparsed_identifier).toSet
+  }
+
+  private def findTables(dag: DAG, plan: ir.LogicalPlan): DAG = {
+    var child = ""
+    var parent = Set.empty[String]
+
+    def collectTables(node: ir.LogicalPlan): Unit = {
+      node match {
+        case n: ir.InsertIntoTable =>
+          child = n.target
+            .find(_.isInstanceOf[ir.NamedTable])
+            .map(_.asInstanceOf[ir.NamedTable].unparsed_identifier)
+            .getOrElse("unknown") // Unknown is when query is a Select Query. May be file name becomes the table name.
+
+        case p: ir.Project => parent = parent ++ getTablName(p)
+
+        case j: ir.Join => parent = parent ++ getTablName(j.left) ++ getTablName(j.right)
+        case _ => // Do nothing for other nodes
+      }
+      node.children.foreach(collectTables)
+    }
+
+    collectTables(plan)
+    val childNode = Node(child)
+    dag.addNode(childNode)
+    parent.foreach { parentTable =>
+      val parentNode = Node(parentTable)
+      dag.addNode(parentNode)
+      dag.addEdge(parentNode, childNode)
```

Comment:
we need more infromation on the edge - e.g. read and write. one query plan will always result in zero or more triplets. 

For example: `SELECT * FROM a LEFT JOIN b RIGHT JOIN c` will result in:
* `a -> read -> b`
* `a -> read -> c`

And `CREATE TABLE a AS SELECT FROM b` will result in:
* `b -> write -> a`

Reply:
in the above example?

For example: SELECT * FROM a LEFT JOIN b RIGHT JOIN c will result in:

`a -> read -> b`
`a -> read -> c`

how are b and c related should there `b -> read -> c` as well
And `CREATE TABLE a AS SELECT FROM b` will result in:

`b -> write -> a`

and in below example  `CREATE TABLE d AS SELECT FROM a LEFT JOIN b RIGHT JOIN c`
Should it be 

`d -> read -> a` or  'a -> write -> d`

Reply:
Done